library/ssl_tls.c

```diff
@@ -6021,11 +6021,31 @@ int mbedtls_ssl_handshake_step( mbedtls_ssl_context *ssl )
 
 #if defined(MBEDTLS_SSL_CLI_C)
     if( ssl->conf->endpoint == MBEDTLS_SSL_IS_CLIENT )
-        ret = mbedtls_ssl_handshake_client_step( ssl );
+    {
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+        if(is_configured_as_tls1_3(ssl->conf))
```

Comment:
We should aim to write this in a way that does not require change when we support runtime-negotiation of TLS 1.2 vs. TLS 1.3.

I would therefore suggest that we introduce a function `ssl_handshake_in_tls12_state_machine()` or (the negative) `ssl_handshake_in_tls13_state_machine()` and to use that for the dispatch here. 

Then, for now we can define those functions purely on the basis of the SSL configuration, as you do in `is_configured_as_tls1_3()`. But when we later add runtime negotiation of 1.2 vs. 1.3, this can be refined.

Does that make sense?

Reply:
I am not fully understand it. And I have create an issue for that.